src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -892,31 +896,55 @@ private class MacroResolver private constructor(private val processor: RsResolve
         // But we want to process macro definition before `bar!` macro call, so we have to use
         // a macro call as a "parent"
         val expandedFrom = (element as? RsExpandedElement)?.expandedFrom
-        if (expandedFrom != null && processExpandedFrom(expandedFrom)) return true
-        val context = expandedFrom ?: element.context ?: return false
+        if (expandedFrom != null && processExpandedFrom(expandedFrom)) return MacroResolveResult.True
+        val context = expandedFrom ?: element.context ?: return MacroResolveResult.False
+
+        if (context is RsMod) {
+            /** [processRemainedExportedMacros] processes local imports */
+            val result = processMacros(context, processor, ::processRemainedExportedMacros)
+            if (result !== null) return result.toResult(usedNewResolve = true)
+        }
+        for (e in element.leftSiblings) {
```

Comment:
This reordering of `processExpandedFrom` and `for (e in element.leftSiblings)` looks breaking in a case like this
```
foo! {
    macro_rules! foo { ... }
    foo!();
}
```